### DISCUSSION TITLE
Mypy screams about not passing a value for validator-default attribute

### DISCUSSION THREAD

**Discussion by mdczaplicki** [pydantic/pydantic#3280] [Question]
I have this schema:

```
class X(BaseModel):
  x: int
  y: int

  @root_validator(pre=True)
  def _set_y(cls, values):
    values["y"] = values["x"] + 2
    return values

x = X(x=2)  # mypy complains here about not passing y
```

Is there anything I can do for mypy not to complain about it? For now I've added `# type: ignore` to initialization of `X`, but maybe there's a way to mark `y` as not mandatory to pass.

**Comment by uriyyo** [comment]
You can use `typing.TYPE_CHECKING` trick:
```py
from typing import TYPE_CHECKING

from pydantic import root_validator, BaseModel


class X(BaseModel):
    x: int

    if TYPE_CHECKING:
        y: int = 0
    else:
        y: int

    @root_validator(pre=True)
    def _set_y(cls, values):
        values["y"] = values["x"] + 2
        return values


x = X(x=2)
```

`mypy` will think that `y` has a default value and will not complain about it.

**Reply by mdczaplicki** [reply]
Looks a litle bit like a code smell :cry:.

**Reply by uriyyo** [reply]
I think it's a common way to handle cases as you have) Here are examples how to `pydantic` uses `TYPE_CHECKING`:
https://github.com/samuelcolvin/pydantic/search?q=if+TYPE_CHECKING

But it's up to you to decide what approach will be better for you
